tools/test_api.py

```diff
@@ -2272,7 +2272,18 @@ def build_tests(tests, base_source_paths, build_path, target, toolchain_name,
 
 def test_spec_from_test_builds(test_builds):
     for build in test_builds:
-        if Target.get_target(test_builds[build]['platform']).is_PSA_non_secure_target:
+        if test_builds[build]['platform'][:12] == 'NU_PFM_M2351':
+            # Translate target names with naming rule below to platform name
+            # 'NU_PFM_M2351' for the NuMaker-PFM-M2351 board:
+            # 'NU_PFM_M2351'+'_'+['P', 'NP']+'_'+['S', 'NS']
+            #
+            # 1. Online database has limit of max 20 chars on target name,
+            #    so we cannot follow the naming rule below for this board:
+            #    'NUMAKER_PFM_M2351'+'_'+['PSA', 'NOPSA']+'_'+['S', 'NS']
+            # 2. The originally registered platform name 'NUMAKER_PFM_M2351'
+            #    in mbed-os-tools must change to 'NU_PFM_M2351' accordingly.
```

Comment:
@ARMmbed/mbed-os-tools  Do we need this hack, can't change the limit in the online database (wonder why is it only 20 char) ?

Reply:
Fundamentally we do not know what the knock-on effect of increasing the database limit might be. The limit has been in place for several years: is there any code that depends on this? The easiest way to resolve this is to keep the board name to 20 characters or fewer, rather than introducing another hack...

Reply:
@ccli8 What can we do about this hack?

Reply:
Looks like the names have now been reduced to fit inside 20 characters, so now ok from our point of view. However, this code is owned by @jamesbeyond so he might well have an opinion on this.

Reply:
Technically this code is not own by my, because this is part of mbed-cli test interface  😉  @mark-edgeworth 

Though, seems the target name already been shortened, I can't see any reason, we need this hard-coding `if` statement to truncate the name, please confirm @ccli8.

Reply:
@jamesbeyond For M2351, the target name is **NU_PFM_M2351_NP_NS** for non-PSA non-secure target or future **NU_PFM_M2351_P_NS** for PSA non-secure target, but the platform name registered in mbed-os-tools is [**NU_PFM_M2351**](https://github.com/ARMmbed/mbed-os-tools/pull/196). The truncation here is to convert target name to mbed-os-tools platform name for Greentea test (written into test spec file). PSA target like **LPC55S69_NS** also has the same truncation.

<pre>
        elif Target.get_target(test_builds[build]['platform']).is_PSA_non_secure_target:
            <b>
            if test_builds[build]['platform'].endswith('_NS'):
                test_builds[build]['platform'] = test_builds[build]['platform'][:-3]
            </b>
            if test_builds[build]['platform'].endswith('_PSA'):
                test_builds[build]['platform'] = test_builds[build]['platform'][:-4]
</pre>

Reply:
Hi @ccli8, I believe the code that you quoting can remove the suffix of `_NS` from your targets. right? so you don't need to worried about the `_NS`. Then how about `_P` or `_NP`? what are they stand for?
In this script we generally not recommend to have code for specific target.  if the `_P` or `_NP` are something general, you can add a line to remove `_P` or `_NP` as well

Reply:
@jamesbeyond Due to max 20 chars limited in online database, `_P` for `PSA` and `_NP` for `_NOPSA`. `_P`/`_NP` are shortened names for M2351 and so target specific code here

Reply:
@ccli8 In this case, `NU_PFM_M2351_PSA_NS` and `NU_PFM_M2351_NS` should be the names for psa and non-psa variants, they both less than 20 char. and they will be handled without extra truncating.

BTW, @MarceloSalazar, do we have naming guidelines for Partners porting their target to mbed-os? to let people know the limitation of 20 chars, and to avoid changing name in the future.

Reply:
@jamesbeyond OK. I will rename them to NU_PFM_M2351_[S, NS] and NU_PFM_M2351_PSA_[S, NS] for non-PSA target and PSA target respectively.

Reply:
We don't have explicit guidelines, but usually recommend to choose a relatively small number of characters to help developers when using the platform, especially with Mbed CLI.
We shouldn't be changing the name at all and this should be considered an exception.

Reply:
@jamesbeyond Split out #11467 to address the name truncation for TZ targets. It is general and no target-specific.